### PR TITLE
Use build args for Dockerfiles

### PR DESCRIPTION
**Pull request by vasily-v-ryabov**
Many our internal Dockerfile's contain "ARG" statements at the beginning of Dockerfile. When we need "docker build", these ARGs are required. The error looks so:
```
+ docker build -t <censored> -f <Dockerfile> .
Sending build context to Docker daemon  9.502MB

Step 1/8 : ARG ARCH
Step 2/8 : ARG DISTRO
Step 3/8 : FROM <censored>
invalid reference format
```

@artemry-mlnx FYI

I think we need better solution in a mid-term so that ARGs are passed more explicitly and transparently. This is just a hotfix.

### CODE REVIEW

**Comment by vasily-v-ryabov** [comment]
I see ARG could be set in `Jenkinsfile`. So maybe this pull request is wrong. Looking into more opinions...

**Review by artemry-mlnx — APPROVED** [review]

File: src/com/mellanox/cicd/Matrix.groovy [hunk 1]
```diff
@@ -366,11 +366,11 @@ Map getMatrixTasks(image, config) {
     return getTasks(axes, image, config, include, exclude)
 }
 
-def buildImage(img, filename, config) {
+def buildImage(img, filename, arch, distro, config) {
```
Comment (mike-dubman) [inline src/com/mellanox/cicd/Matrix.groovy:369]: best to have:
```
buildImage(img, file, build_args, config
```
code should iterate over ```arg_list``` and convert it to string and prepend ```-build-args```

File: src/com/mellanox/cicd/Matrix.groovy [hunk 2]
```diff
@@ -433,7 +433,7 @@ def buildDocker(image, config) {
             }
             if (need_build) {
                 config.logger.info("Building - ${img} - ${filename}")
-                buildImage(img, filename, config)
+                buildImage(img, filename, arch, distro, config)
```
Comment (mike-dubman) [inline src/com/mellanox/cicd/Matrix.groovy:436]:
```buildImage(img, filename, ["ARCH=$arch", "DISTRO=$distro"], config)

**Review by mike-dubman — CHANGES_REQUESTED** [review]
lets make it more general if will need to pass additional build arg args

**Comment by vasily-v-ryabov** [comment]
bot:retest

File: src/com/mellanox/cicd/Matrix.groovy [hunk 3]
```diff
@@ -405,10 +405,11 @@ String getChangedFilesList(config) {
 
 def buildDocker(image, config) {
 
-    def img      = image.url
-    def arch     = image.arch
-    def filename = image.filename
-    def distro   = image.name
+    def img = image.url
+    def arch = image.arch
+    def filename = image.filename.toString()
```
Comment (mike-dubman) [inline src/com/mellanox/cicd/Matrix.groovy:410]: ```image.filename.toString().trim()``` to cut trailing spaces
why toString needed at all?

File: src/com/mellanox/cicd/Matrix.groovy [hunk 4]
```diff
@@ -427,13 +428,15 @@ def buildDocker(image, config) {
                 config.logger.info("Forcing building file per user request: ${filename} ... ")
                 need_build++
             }
-            if (config.get("cFiles").contains(filename)) {
+            config.logger.debug("Dockerfile name: '${filename}'")
```
Comment (mike-dubman) [inline src/com/mellanox/cicd/Matrix.groovy:431]: IMO not needed extra quotes in ``` 'str'```, it is bad for copy&paste

File: src/com/mellanox/cicd/Matrix.groovy (hunk 3 above)
Comment (vasily-v-ryabov) [reply src/com/mellanox/cicd/Matrix.groovy:410]: It is needed to pass condition `if (config.get("cFiles").contains(filename))`. The dockerfiles were changed, but re-built has not been happened. I will comment in the code that solution was found here: https://stackoverflow.com/q/56829842/3648361

Comment (vasily-v-ryabov) [reply src/com/mellanox/cicd/Matrix.groovy:410]: It works without `.trim()`, but I can add it.

File: src/com/mellanox/cicd/Matrix.groovy (hunk 4 above)
Comment (vasily-v-ryabov) [reply src/com/mellanox/cicd/Matrix.groovy:431]: Sure, I've just tried to detect extra spaces. In fact it's not the case. I'll remove single quotes.

**Review by mike-dubman — APPROVED** [review]

**Pull request merged**